pubsub/README.rst

```diff
@@ -19,174 +19,27 @@ Usage
+At the moment, ``gcloud-{aio,rest}-pubsub`` only provides a set of handles
+to call Pubsub REST API.
 
-An HTTP-oriented version, in keeping with the other ``gcloud-{aio,rest}-*`` libraries,
-will likely be coming soon -- though our current approach works reasonably well
-for allowing the official ``grpc`` client to be used under ``asyncio``, we
-continue to see threading oddities now and again which we've not been able to
-solve. As such, we do not wholeheartedly recommend using the
-``SubscriberClient`` of this library in production, though a resilient enough
-environment for your use-case may be possible.
-
-Here's the rough usage pattern for subscribing:
+Here's the rough usage pattern for using ``SubscriberClient``:
 
 .. code-block:: python
 
     from gcloud.aio.pubsub import SubscriberClient
     from gcloud.aio.pubsub import SubscriberMessage
 
     client = SubscriberClient()
-    # create subscription if it doesn't already exist
-    client.create_subscription(
+    # create subscription
+    await client.create_subscription(
         'projects/<project_name>/subscriptions/<subscription_name>',
         'projects/<project_name>/topics/<topic_name>')
 
-    async def message_callback(message: SubscriberMessage) -> None:
-        try:
-            # just an example: process the message however you need to here...
-            result = handle(message)
-            await upload_result(result)
-        except Exception:
-            message.nack()
-        else:
-            message.ack()
-
-    # subscribe to the subscription, receiving a Future that acts as a keepalive
-    keep_alive = client.subscribe(
-        'projects/<project_name>/subscriptions/<subscription_name>',
-        message_callback)
-
-    # have the client run forever, pulling messages from this subscription,
-    # passing them to the specified callback function, and wrapping it in an
-    # asyncio task.
-    client.run_forever(keep_alive)
-
-Configuration
-^^^^^^^^^^^^^
-
-Our create_subscription method is a thin wrapper and thus supports almost all
-keyword configuration arguments from the official pubsub client which you can
-find in the `official Google documentation`_.
-
-Since the underlying Google implementation of ``Scheduler`` only allows for the
-concrete ``ThreadScheduler`` which is also the default, we have opted not to
-expose this configuration option. Additionally, we would like to fully
-deprecate said Google implementation in favour of a fully ``asyncio``
-implementation which uses the event loop as the scheduler.
-
-When subscribing to a subscription you can optionally pass in a ``FlowControl``
-instance.
-
-.. code-block:: python
-
-    example_flow_control = FlowControl(
-        max_bytes=100*1024*1024,
-        max_messages=1,
-        max_lease_duration=10,
-        max_duration_per_lease_extension=0,
-    )
-
-    keep_alive = client.subscribe(
+    # pull messages
+    await client.pull(
```

Comment:
Maybe you want to include here a variable to catch the return value to demonstrate that you now pull messages directly? Something like this?

```python
messages: List[SubscriberMessage] = await client.pull(...)
```